Issue1389 detailed house

LoneMeertens opened the pull request:
Fixes #1389.
More detailed documentation and result visualization can be added if deemed valuable. This can be combined with open-ideas/CrashCourse#27.

lucasverleyen commented:
@LoneMeertens, thank you for addressing the issue! I really like the new structure, a great improvement. :-) Also, adding more documentation/explanation provides significant added value. 
I suggest including https://github.com/open-ideas/__CrashCourse__/issues/25 and https://github.com/open-ideas/IDEAS/issues/1374 as well in this PR, which seems more convenient than having too many issues and PRs related to the same models. 
And regarding more detailed documentation and result visualisation, I suggest maintaining consistency with the SimpleHouse tutorial from IBPSA and following a similar approach. This tutorial has been thoroughly reviewed recently and is consequently in good shape.

LoneMeertens commented:
@lucasverleyen Thank you for the detailed feedback and suggestions! 😊 I'm glad you like the new structure and the additional documentation.

I’ll take this up and include open-ideas/__CrashCourse__#25 and Update Tutorial IDEAS #1374 in this PR as suggested—it does make sense to consolidate related issues and PRs.

Regarding the documentation and result visualization, I'll align it with the SimpleHouse tutorial from IBPSA to ensure consistency and maintain the high standard of quality. I'll also add DetailedHouse10 (as DetailedHouse) as a final example, positioned at the same level as SimpleHouse.

Thanks again for the valuable input!

lucasverleyen commented:
@jelgerjansen good catch (__Dymola_Commands)! I suggest adding this to the documentation of the testing framework: https://github.com/open-ideas/IDEAS/blob/master/IDEAS/Resources/Scripts/tests/README.md under Developing new tests. Something like: "Make sure that the file name and path in the model annotation under __Dymola_Comands correspond to the name and path to the .mos file in Resources/Scripts/Dymola/... " What do you think?

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/package.mo:
```diff
@@ -0,0 +1,107 @@
+within IDEAS.Examples.Tutorial;
+package DetailedHouse "Package with example for how to built a simple house, using the detailed building envelope component
```
I would remove the "simple" as it is in contradiction with "DetailedHouse".

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/package.mo:
```diff
@@ -0,0 +1,107 @@
+within IDEAS.Examples.Tutorial;
+package DetailedHouse "Package with example for how to built a simple house, using the detailed building envelope component
+  models, HVAC and other components"
```
"Package with example for how to built an office building (using the detailed building envelope component models within IDEAS.Buildings) with occupants, a radiator heating system connected to a heat pump, and a ventilation system."

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/package.mo:
```diff
@@ -0,0 +1,107 @@
+  annotation (Documentation(info="<html>
+<p>
+This package contains examples with step-by-step instructions for how to build a system model
+for a simple house with detailed building envelope component models, HVAC and other components.
```
Also update this line according to the comment above.

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/package.mo:
```diff
@@ -0,0 +1,107 @@
+
+  annotation (Documentation(info="<html>
+<p>
+This package contains examples with step-by-step instructions for how to build a system model
```
The step-by-step instructions are currently not fully included in the documentation of each model, thus I would either add this or clearly refer to the instructions in the crash course repo (https://github.com/open-ideas/__CrashCourse__/blob/master/Exercises/Exercise%203/Latex/Exercise3.pdf and https://github.com/open-ideas/__CrashCourse__/blob/master/Exercises/Exercise%203/Latex/Exercise4.pdf).

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/package.mo:
```diff
@@ -0,0 +1,107 @@
+the use of equations falls outside of the scope of this exercise.
+</p>
+<p>
+For this exercise you will create a model of a simple house,
+consisting of a heating system, detailed building envelope component models, and a ventilation model.
```
Be more specific: Firstly, develop a single-zone (office) building using the component models of the IDEAS.Buildings package, then add occupants. Secondly, move to a two-zone (office) building and add a heating system and a ventilation system (including their control).

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/package.mo:
```diff
@@ -0,0 +1,107 @@
+<p>
+For this exercise you will create a model of a simple house,
+consisting of a heating system, detailed building envelope component models, and a ventilation model.
+The exercise starts from a template file that should not produce any errors.
```
This exercise does not have a template, does it? (There is no DetailedHouse0.mo file as it appears in the SimpleHouse tutorial).

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/package.mo:
```diff
@@ -0,0 +1,107 @@
+For this exercise you will create a model of a simple house,
+consisting of a heating system, detailed building envelope component models, and a ventilation model.
+The exercise starts from a template file that should not produce any errors.
+This file will be extended in several steps, adding complexity.
```
It exists of two parts: 1) single-zone and occupancy, 2) two-zone and HVAC with two different starting points (DetailedHouse1 and DetailedHouse5)

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/package.mo:
```diff
@@ -0,0 +1,107 @@
+<a href=\"modelica://IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse1\">
+IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse1</a>
+represents a basic building model that includes a single zone, four walls, a window, a floor, and a ceiling
+and serves as a starting model to implement the entire <code>DetailedHouse</code> model.
```
I wouldn't say "entire" as `DetailedHouse5` serves as another starting point. This model is based on `DetailedHouse1`, but not exactly the same (as it uses two rectangular zone models instead of the separate building envelope models).

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/package.mo:
```diff
@@ -0,0 +1,107 @@
+<li>
+<a href=\"modelica://IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse5\">
+IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse5</a>
+This example teaches the use of the <code>RectangularZoneTemplate</code> .
```
"shows the use" instead of "This example teaches the use".

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/package.mo:
```diff
@@ -0,0 +1,107 @@
+<li>
+<a href=\"modelica://IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse6\">
+IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse6</a>
+adds a HVAC system. The system consists of a water-water heat pump, radiators, a storage tank, circulation
```
"adds a heating system". The ventilation is not yet added, so don't use HVAC. (FYI, it's "an HVAC" instead of "a HVAC")

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/package.mo:
```diff
@@ -0,0 +1,107 @@
+<a href=\"modelica://IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse6\">
+IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse6</a>
+adds a HVAC system. The system consists of a water-water heat pump, radiators, a storage tank, circulation
+pumps and a low temperature heat source for the heat pump.
```
", and a low-temperature heat source at constant temperature"

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/package.mo:
```diff
@@ -0,0 +1,107 @@
+<a href=\"modelica://IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse7\">
+IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse7</a>
+adds a controller that disables the heat pump when the supply water
+temperature exceeds 45◦C.
```
`<i>45°C</i>`

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/package.mo:
```diff
@@ -0,0 +1,107 @@
+<a href=\"modelica://IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse8\">
+IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse8</a>
+adds workflow automation, generating a CSV file at a userdefined location that contains data 
+for each of the inputs of the block .
```
Remove abundant whitespace between "block" and "."

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/package.mo:
```diff
@@ -0,0 +1,107 @@
+<li>
+<a href=\"modelica://IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse9\">
+IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse9</a>
+adds a CO2-controlled ventilaation system. The occupancy model from <a href=\"modelica://IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse4\">
```
Typo: "ventilation"

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse9.mo:
```diff
@@ -165,7 +163,39 @@ First implementation for the IDEAS crash course.
+IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse4</a> is added to one zone and a 
+fixed occupancy of 1 person to the other zone. The ventilation system
+consists of two fans, two supply and two return air VAVs (Variable Air Volume), a heat recovery unit and an
+outdoor air source. The control consists of PI controllers with a set point of 1000 ppm.
```
setpoint (maybe also apply this change to the exercise sheet of the crash course)

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/package.mo:
```diff
@@ -0,0 +1,107 @@
+IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse4</a> is added to one zone and a 
+fixed occupancy of 1 person to the other zone. The ventilation system
+consists of two fans, two supply and two return air VAVs (Variable Air Volume), a heat recovery unit and an
+outdoor air source. The control consists of PI controllers with a set point of 1000 ppm.
```
setpoint (maybe also apply this change to the exercise sheet of the crash course)

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/package.mo:
```diff
@@ -0,0 +1,107 @@
+<li>
+<a href=\"modelica://IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse10\">
+IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse10</a>
+removes small time constants from the problem to decrease computation time.
```
"removes small time constants from the model"

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/package.mo:
```diff
@@ -0,0 +1,107 @@
+<p>
+For each stage, firstly the model part is qualitatively explained.
+Next, the names of the required Modelica models (from the Modelica Standard Library and/or IDEAS library) are listed.
+Finally, we provide high-level instructions of how to set up the model.
```
I don't think this is done for DetailedHouse (it is for SimpleHouse)

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/package.mo:
```diff
@@ -0,0 +1,107 @@
+Finally, we provide high-level instructions of how to set up the model.
+If these instructions are not clear immediately, have a look at the model documentation and at the type of connectors the model has,
+try out some things, make an educated guess, etc.
+Finally, we provide reference results that allow you to check if your implementation is correct.
```
Also not included for DetailedHouse

jelgerjansen reviewed (changes requested):
@LoneMeertens thank you for addressing this issue. Most comments are related to model documentation and syntax. 

The models previously didn't run due to the reference to an unexisting mos file in `__Dymola_Commands()`. This does not generates warnings when translating the model in Dymola, but it does for the unit tests. I solved this issue via commits 8cca50086c3e319f3d3e5d05645a847f93f34b22 and e8f49c53f9c8fe47e1a212c29f234f2230d64ae2

Once you've also addressed the other related issues (e.g. #1374), I'll do another review.

LoneMeertens commented:
@jelgerjansen @lucasverleyen I have addressed the mentioned comments and added a 'Connection Instructions' and 'Reference Result' section to each sub DetailedHouse model, to align with the SimpleHouse example. Additionally, the related issues (e.g., https://github.com/open-ideas/IDEAS/issues/1374) have been resolved. Thank you for reviewing this! Please let me know if there are any minor adjustments needed, and I will be happy to make them.

lucasverleyen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse1.mo:
```diff
@@ -0,0 +1,200 @@
+IDEAS.Buildings.Components.Window</a>
+</li>
+<li>
+<a href=\"modelica://IDEAS.Buildings.Components.InternalWall\">
```
As far as I remember, we decided last year to remove the internal wall representing the floor and ceiling and use a slab on ground for the floor and another outer wall for the ceiling. This was indeed not the case in this tutorial, but I would change the tutorial according to the changes of last year in the crash course exercise

lucasverleyen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse1.mo:
```diff
@@ -0,0 +1,200 @@
+<img alt=\"Zone temperature as function of time.\"
+src=\"modelica://IDEAS/Resources/Images/Examples/Tutorial/DetailedHouse/DetailedHouse1.png\" width=\"700\"/>
+</p>
+</html>", revisions="<html>
```
You can update the revision history of each model, with a reference to this pull request or the issue

lucasverleyen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse3.mo:
```diff
@@ -0,0 +1,62 @@
+</ul>
+<h4>Connection instructions</h4>
+<p>
+Set the appropriate replaceable models in the zone model.
```
From the last crash course, I remember that it was not clear that you could simply set this in the zone model's dialogue window. Maybe you can specify this in this sentence by adding "dialogue window"?

jelgerjansen replied:
This is to solve https://github.com/open-ideas/__CrashCourse__/issues/25

lucasverleyen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse9.mo:
```diff
@@ -165,7 +163,76 @@ First implementation for the IDEAS crash course.
 </html>", info="<html>
 <p>
-Adding CO2-controlled ventilation system.
+Adding CO2-controlled ventilation system. The occupancy model
```
```suggestion
Adding CO<sub>2</sub>-controlled ventilation system. The occupancy model
```

lucasverleyen replied:
can you add sub- and superscripts for CO2 and the units (m2, m3, ...)

lucasverleyen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse9.mo:
```diff
@@ -165,7 +163,76 @@ First implementation for the IDEAS crash course.
+IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse4</a> is added to one zone and a 
+fixed occupancy of 1 person to the other zone. The ventilation system
+consists of two fans, two supply and two return air VAVs (Variable Air Volume), a heat recovery unit and an
+outdoor air source. The control consists of PI controllers with a setpoint of 1000 ppm.
```
```suggestion
outdoor air source. The control consists of PI controllers with a setpoint of 1000 <i>ppm</i>.
```

lucasverleyen commented:
@LoneMeertens thanks a lot for all improvements! I have added a few more comments. I see there are still some comments by Jelger that are open. The easiest is to click on "Resolve conversation" when you have addressed the comment, and then it is clear which comments have been addressed.

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse1.mo:
```diff
@@ -117,8 +116,34 @@ equation
+This first example file instantiates a simple building model that consists of one zone, four walls,
+a window, a floor and a ceiling.  The zone dimensions are 8 m (with walls oriented 
+north and south) by 4 m, and the window measures 3 m by 1.4 m. Use the default 
+zone height of 2.8 m. Apply double glazing and a heavy wall, which provide high thermal mass.
```
The exact types are mentioned in the connection instructions.

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse1.mo:
```diff
@@ -0,0 +1,200 @@
+
+<h4>Reference result</h4>
+<p>
+This model is simulated with the following settings:
```
Include the solver and tolerance that was used.

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse1.mo:
```diff
@@ -0,0 +1,200 @@
+declarations to point towards the higher level <code>SimInfoManager</code> declaration. This way all model equations for
+the weather data have to be generated only once, instead of for each surface.
+</p>
+
```
Remove whitespace

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse5.mo:
```diff
@@ -78,14 +78,51 @@ First implementation for the IDEAS crash course.
+window placement has on the zone dynamics!
+</p>
+<p align=\"center\">
+<img alt=\"Zone temperature for the zone with the north oriented window (blue) and the zone with the south
+oriented window (red).\"
```
This information is not visible in the documentation view. Consider copying/moving this (and all uses of `img alt` in previous and future models) to the Reference results section.

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse5.mo:
```diff
@@ -78,14 +78,51 @@ First implementation for the IDEAS crash course.
+</p>
+<h4>Reference result</h4>
+<p>
+The figure bellow shows the zone temperatures of both zones. Note the large influence that the
```
"effect" instead of `influence`

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse5.mo:
```diff
@@ -1,5 +1,5 @@
+model DetailedHouse5 "New model with 2 coupled zones"
   extends Modelica.Icons.Example;
   replaceable package Medium = IDEAS.Media.Air "Air medium";
 
```
- Change zone height to 2.8m
- Remove obsolete whitespace at line 10 (sim)
- Rename instances, e.g. `zon1` and `zon2` or `recZon1` and `recZon2`

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse6.mo:
```diff
@@ -1,6 +1,7 @@
+  extends DetailedHouse5;
   package MediumWater = IDEAS.Media.Water "Water Medium";
 
   Fluid.HeatPumps.ScrollWaterToWater heaPum(
```
Rename pump instances (e.g. `pumPri` and `pumSec`) and constant for HP control (e.g. `heaPumOn`)

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse6.mo:
```diff
@@ -173,8 +178,104 @@ September 18, 2019 by Filip Jorissen:<br/>
+info="<html>
+<p>
+This model extends <a href=\"modelica://IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse5\">
+IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse5</a> by adding a HVAC system.
```
an HVAC system

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse6.mo:
```diff
@@ -173,8 +178,104 @@ September 18, 2019 by Filip Jorissen:<br/>
+IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse5</a> by adding a HVAC system.
+The system consists of a water-water heat pump, radiators, a storage tank, circulation
+pumps and a low temperature heat source for the heat pump. The model includes constant 
+control set points for the heat pump and pumps. A system is incorporated to measure 
```
```suggestion
control setpoints for the heat pump and pumps. An integrator block is incorporated to measure 
```

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse6.mo:
```diff
@@ -173,8 +178,104 @@ September 18, 2019 by Filip Jorissen:<br/>
+This model extends <a href=\"modelica://IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse5\">
+IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse5</a> by adding a HVAC system.
+The system consists of a water-water heat pump, radiators, a storage tank, circulation
+pumps and a low temperature heat source for the heat pump. The model includes constant 
```
```suggestion
pumps and a heat source at a constant temperature of <i>10°C</i> for the heat pump. The model includes constant 
```

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse6.mo:
```diff
@@ -173,8 +178,104 @@ September 18, 2019 by Filip Jorissen:<br/>
+</li>
+</ul>
+<h4>Connection instructions</h4>
+<p>
+The model components of <a href=\"modelica://IDEAS.Buildings\"> IDEAS.Buildings</a> have been
+integrated where possible. E.g. a zone can always have an occupancy schedule, or lighting models. However,
+for HVAC, many configurations exist in reality. Some zones have a radiator, which physically resides within the
+zone, while others may use floor heating, which is a property of the floor. Some zones may even have multiple
+radiators with different sizes, etc. Due to this multitude of options, HVAC components have not been integrated
+into the IDEAS.Buildings components (yet). Instead, they can be added manually by the user. This requires
+some knowledge by the user about how to interconnect the different components, and even about the relevant
+physical principles that are at play.
+</p>
```
I would remove this section, as it has nothing to do with connection instructions.

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse6.mo:
```diff
@@ -173,8 +178,104 @@ September 18, 2019 by Filip Jorissen:<br/>
+some knowledge by the user about how to interconnect the different components, and even about the relevant
+physical principles that are at play.
+</p>
+<p>
+One notable example is that in each fluid loop the <i>absolute</i> pressure of that loop has to be
+defined somewhere: pumps and valves only provide information about <i>differential</i> pressures. For this purpose
+use <a href=\"modelica://IDEAS.Fluid.Sources.Boundary_pT\"> IDEAS.Fluid.Sources.Boundary_pT</a> and connect it to the loop, 
+which will set the absolute pressure at the connection point.
+</p>
```
Add this directly after the sentence "A reference implementation..."

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse6.mo:
```diff
@@ -173,8 +178,104 @@ September 18, 2019 by Filip Jorissen:<br/>
+<h4>Reference result</h4>
+This model is simulated with the following settings:
+</p>
+<ol>
+<li>Start time = 1e7,</li>
+<li>Stop time = 1.1e7,</li>
+<li>Number of intervals = 5000.</li>
+</ol>
```
You didn't include this for all models, did you? If not, I would only add it in the first example, and store the settings in the model itself.

LoneMeertens replied:
@jelgerjansen I applied this only to the first and sixth models, as they correspond to the starting points of exercises 3 and 4 in the Crash Course, respectively. I plan to keep this block exclusively in the first model, as suggested. How can I store the settings for the other models?

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse6.mo:
```diff
@@ -173,8 +178,104 @@ September 18, 2019 by Filip Jorissen:<br/>
+<li>Stop time = 1.1e7,</li>
+<li>Number of intervals = 5000.</li>
+</ol>
+<p>
+The simulation starts 10<sup>7</sup> seconds after New Year and ends 10<sup>6</sup> seconds later, covering a period of 11.6 days. 
+<p>
```
Remove this

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse7.mo:
```diff
@@ -0,0 +1,85 @@
+  connect(senTemSup.T, hys.u) annotation (Line(points={{136,49},{136,4},{120,4},
+          {120,-40},{40,-40},{40,-70},{58,-70}}, color={0,0,127}));
+  annotation (
+    Documentation(revisions="<html>
+<ul>
+<li>
+September 18, 2019 by Filip Jorissen:<br/>
+First implementation for the IDEAS crash course.
+</li>
+</ul>
```
Move this to end of annotation section.

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse7.mo:
```diff
@@ -0,0 +1,85 @@
+</html>", info="<html>
+<p>
+This step adds a controller that disables the heat pump when the supply water
+temperature exceeds <i>45°C</i>. The simple controller has a large impact on the heat pump COP.
```
"heat pump's COP"

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse7.mo:
```diff
@@ -0,0 +1,85 @@
+about <i>0.25°C</i>. The COP however increases from about <i>2.9</i> to about <i>3.1</i>. Consequently, the energy use over the
+period is reduced from <i>11.56 kWh</i> to <i>9.64 kWh</i>. Note that this heating system configuration is still not efficient
+since the small flow rates still cause large temperatures to occur within the heat pump and thus cause a small
+COP. COP’s of more than 5 are obtainable when using a bypass and a separate pump to charge the storage tank.
```
COPs

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse7.mo:
```diff
@@ -0,0 +1,85 @@
+The figure below compares the results with the results without controller of <a href=\"modelica://IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse6\">
+IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse6</a>. We see that indeed,
+the supply temperature is reduced significantly. This causes the zone temperature to be slightly lower, up to
+about <i>0.25°C</i>. The COP however increases from about <i>2.9</i> to about <i>3.1</i>. Consequently, the energy use over the
```
The increase in COP is relatively small. @lucasverleyen do we still want to change this (see https://github.com/open-ideas/__CrashCourse__/issues/19)?
The energy use reduces by +/- 2 kWh (about 15%), which is significant I would say.

jelgerjansen replied:
I will change the documentation such that it focuses on the electricity use

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse8.mo:
```diff
@@ -0,0 +1,63 @@
+equation
+  connect(jsonWri.u[1], EEl.y) annotation (Line(points={{280,-32},{286,-32},{286,
+          50},{301,50}},     color={0,0,127}));
+  annotation (Documentation(revisions="<html>
+<ul>
+<li>
+September 18, 2019 by Filip Jorissen:<br/>
+First implementation for the IDEAS crash course.
+</li>
+</ul>
```
Move revision history to end of annotation section.

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse8.mo:
```diff
@@ -0,0 +1,63 @@
+</ul>
+</html>", info="<html>
+<p>
+Extracting results from Modelica/Dymola can be tedious. Therefore several custom tools have been developed to facilitate 
```
```suggestion
Extracting results from Dymola can be tedious. Therefore, several custom tools have been developed to facilitate 
```

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse8.mo:
```diff
@@ -0,0 +1,63 @@
+<a href=\"modelica://Modelica.Blocks.Sources.CombiTimeTable\"> Modelica.Blocks.Sources.CombiTimeTable</a>.
+</p>
+<p>
+In this example we will not output time series data, but instead a single value: the total energy use at the end
```
```suggestion
In this example we will not output time series data, only a single value: the total energy use at the end
```

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse8.mo:
```diff
@@ -0,0 +1,63 @@
+      StopTime=11000000,
+      __Dymola_NumberOfIntervals=5000,
+      Tolerance=1e-06,
+      __Dymola_fixedstepsize=20,
```
Remove this line (refers to Euler integrator, which is not used here)

jelgerjansen commented on IDEAS/Resources/Scripts/Dymola/Examples/Tutorial/DetailedHouse/DetailedHouse9.mos:
```diff
@@ -1,4 +1,4 @@
-simulateModel("IDEAS.Examples.Tutorial.Example10", startTime=10000000, stopTime=11000000, numberOfIntervals=5000, method="Euler", fixedstepsize=20, resultFile="Tutorial10");
+simulateModel("IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse9", startTime=10000000, stopTime=11000000, numberOfIntervals=5000, method="LSodar", tolerance=1e-4, resultFile="DetailedHouse9");
```
Change tolerance to 1e-6

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse7.mo:
```diff
@@ -0,0 +1,85 @@
+      StopTime=11000000,
+      __Dymola_NumberOfIntervals=5000,
+      Tolerance=1e-06,
+      __Dymola_fixedstepsize=20,
```
Remove this line (refers to Euler integrator, which is not used here)

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse9.mo:
```diff
@@ -11,7 +11,6 @@ model Example9 "Adding CO2-controlled ventilation"
-
   Fluid.Actuators.Dampers.PressureIndependent vavSup(
     redeclare package Medium = Medium,
     m_flow_nominal=100*1.2/3600,
```
Rename instance name OutsideAir to `outAir`

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse1.mo:
```diff
@@ -0,0 +1,200 @@
+within IDEAS.Examples.Tutorial.DetailedHouse;
+model DetailedHouse1 "First example model containing one zone"
+  extends Modelica.Icons.Example;
+  package Medium = IDEAS.Media.Air "Air medium";
```
Suggestion to rename "Medium" to "MediumAir" to clearly distinct from "MediumWater" (which will be used later in the tutorial).

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse10.mo:
```diff
@@ -49,7 +49,11 @@ record HeavyWall "BESTEST heavy wall with idealized wood layer"
+ experiment(
       StartTime=10000000,
       StopTime=11000000,
       __Dymola_NumberOfIntervals=5000,
```
Remove the tolerance as this is not used in Example 10 (Euler integrator)

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/package.mo:
```diff
@@ -0,0 +1,128 @@
+</p>
+<p>
+A more detailed explanation and step-by-step approach are provided in Exercises 3 and 4 
+of the Modelica Crash Course, developed by the SySi team at KU Leuven. Links to these 
```
"The SySi Team" (capitals)

jelgerjansen reviewed (changes requested):
@LoneMeertens thanks a lot for all your work!
I still have quite some remarks and suggestions, a lot of them focusing on lay-out and naming conventions. Please let me know if you don't find time to resolve these, then I'll have a look at it.

LoneMeertens commented:
@jelgerjansen @lucasverleyen All requested changes have been committed. Please let me know if any further adjustments are needed. Thank you for reviewing these updates.

lucasverleyen commented on IDEAS/Resources/Images/Examples/Tutorial/DetailedHouse/Example6.png:
Ideally, you could change the name to DetailedHouse6.png, to be consistent with previous images.

lucasverleyen commented on IDEAS/Resources/Images/Examples/Tutorial/DetailedHouse/Example6_bis.png:
Ideally, you could change the name to DetailedHouse6_bis.png, to be consistent with previous images.

lucasverleyen commented on IDEAS/Resources/Images/Examples/Tutorial/DetailedHouse/Example7.png:
Ideally, you could change the name to DetailedHouse7.png, to be consistent with previous images.

lucasverleyen commented on IDEAS/Resources/Images/Examples/Tutorial/DetailedHouse/Example9.png:
Ideally, you could change the name to DetailedHouse9.png, to be consistent with previous images.

lucasverleyen commented on IDEAS/Resources/Images/Examples/Tutorial/DetailedHouse/Example9_bus.png:
Ideally, you could change the name to DetailedHouse9_bis.png, to be consistent with previous images.

lucasverleyen commented on IDEAS/Resources/Images/Examples/Tutorial/DetailedHouse/Schematic6.png:
For the schematics, maybe DetailedHouse6_schematic.png, to be consistent with previous images?

lucasverleyen commented on IDEAS/Resources/Images/Examples/Tutorial/DetailedHouse/Schematic7.png:
For the schematics, maybe DetailedHouse7_schematic.png, to be consistent with previous images?

lucasverleyen commented on IDEAS/Resources/Images/Examples/Tutorial/DetailedHouse/Schematic9.png:
For the schematics, maybe DetailedHouse9_schematic.png, to be consistent with previous images?

lucasverleyen reviewed (commented):
@LoneMeertens, thanks for all the changes, huge improvements! :-)
I have added a few comments on the image file names. And I have taken the freedom to improve some graphics and documentation myself. I'll have a detailed look at DetailedHouse9 and 10 before the beginning of next week, as it is too late now...

jelgerjansen commented:
I did some more updates, which resolves all previously mentioned 

To do:
1) Change exercise 7: use pump to decouple the flow rates in the primary and secondary side of the network to avoid chattering (which occurs due to a combination of an on/off HP and too small flow rates through the radiators (and therefore the heat pump)). A similar control approach is used in the CONTROL exercise session of the Thermal Systems course
2) Update figures and quantitative results in Modelica documentation.
3) Update reference results (for the unit tests)
4) Update figures and quantitative results in the Crash course material (https://github.com/open-ideas/__CrashCourse__) 

@lucasverleyen and @LoneMeertens unless you would like to spend time on this during the weekend, let's discuss this further on Monday.

annadellisola commented:
The hydraulic scheme in IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse6 should be changed to decouple the heat pump operation from the emission system.

annadellisola commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse10.mo:
The Euler solver was not working due to fast dynamics in the **pumEmi**. The issue was solved in commit ad2e25ba1224c84e957bcb237cc228584ce3c281

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse10.mo:
```diff
@@ -0,0 +1,128 @@
+can significantly increase for larger simulation time due to frequent on/off switching 
+of the heat pump. These effects cause a lot of fast transients that force the solver
+to take small steps, which takes a lot of time.
+<p>
```
</p>
<p>

jelgerjansen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse6.mo:
```diff
@@ -0,0 +1,277 @@
+use <a href=\"modelica://IDEAS.Fluid.Sources.Boundary_pT\"> IDEAS.Fluid.Sources.Boundary_pT</a> and connect it to the loop, 
+which will set the absolute pressure at the connection point.
+</p>
+<p>
```
Obsolete: remove this

lucasverleyen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse8.mo:
```diff
@@ -0,0 +1,65 @@
+</p>
+<p>
+In this example we will not output time series data, only a single value: the total electricity consumption at the end
+of the simulation, using the <code>JSON</code> file format. Note that the external library <code>ExternData</code> can be used to read
```
```suggestion
of the simulation, using the <code>JSON</code> file format. Note that the external library <a href="https://build.openmodelica.org/Documentation/ExternData.html">ExternData</a> can be used to read
```

lucasverleyen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse9.mo:
```diff
@@ -0,0 +1,240 @@
+        "Simulate and plot"),
+    Documentation(info="<html>
+<p>
+Adding CO<sub>2</sub>-controlled ventilation system. The occupancy model
```
```suggestion
This example adds a CO<sub>2</sub>-controlled ventilation system. The occupancy model
```

lucasverleyen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse9.mo:
```diff
@@ -0,0 +1,240 @@
+<h4>Required models</h4>
+<ul>
+<li>
+Custom occupant model from <a href=\"modelica://IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse4\">
```
```suggestion
Custom occupancy model from <a href=\"modelica://IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse4\">
```

lucasverleyen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse9.mo:
```diff
@@ -0,0 +1,240 @@
+<h4>Connection instructions</h4>
+<p>
+This model extends from <a href=\"modelica://IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse8\">
+IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse8</a> where its existing medium declaration is modified to add CO2.
```
```suggestion
IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse8</a> where its existing medium declaration is modified to add CO<sub>2</sub>.
```

lucasverleyen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse9.mo:
```diff
@@ -0,0 +1,240 @@
+</p>
+<p>
+The model includes two PI controllers, with their outputs connected to the VAVs. The zone <code>ppm</code> 
+outputs are connected to the measurement inputs <code>u_m</code> of the PI controllers, and a constant set point 
```
```suggestion
outputs are connected to the measurement inputs <code>u_m</code> of the PI controllers, and a constant setpoint 
```

lucasverleyen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse9.mo:
```diff
@@ -0,0 +1,240 @@
+</p>
+<h4>Reference result</h4>
+<p>
+The figures below show the operative zone temperature, CO2 concentrations and PI control signals in both zones.
```
```suggestion
The figures below show the operative zone temperature, CO<sub>2</sub> concentrations and PI control signals in both zones.
```

lucasverleyen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse9.mo:
```diff
@@ -0,0 +1,240 @@
+IDEAS.Examples.Tutorial.DetailedHouse.DetailedHouse4</a> is added to one zone and a 
+fixed occupancy of 1 person to the other zone. The ventilation system
+consists of two fans, two supply and two return air VAVs (Variable Air Volume), a heat recovery unit and an
+outdoor air source. The control consists of PI controllers with a setpoint of 1000 <i>ppm</i>.
```
```suggestion
outdoor air source. The control consists of PI controllers with a setpoint of <i>1000 ppm</i>.
```

lucasverleyen replied:
As far as I can see, you use the following rules:
- number + abbreviated unit: both italic
- only number: normal font, no italic
- number + unit in full form (e.g. days, persons...): normal font

lucasverleyen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse6.mo:
```diff
@@ -0,0 +1,270 @@
+<p>
+This example illustrates the importance of control, which is currently not modelled. All pumps and the heat
+pump are assumed to be active continuously, which is detrimental for the system performance. The COP
+(heaPum.com.COP) is only about 2.5.
```
```suggestion
(heaPum.com.COP) is only about <i>2.5</i>.
```

lucasverleyen replied:
you use italic font for COP values in example 6

lucasverleyen commented on IDEAS/Examples/Tutorial/DetailedHouse/DetailedHouse9.mo:
```diff
@@ -0,0 +1,240 @@
+IDEAS.Fluid.Sources.Boundary_pT</a> except that it automatically sets the outdoor dry bulb temperature and humidity.
+</p>
+<p>
+For this exercise, assume that the VAV has a nominal flow rate of <i>100 m3/h</i>, which equals <i>0.033 kg/s</i>. A nominal pressure drop of <i>50 Pa</i>
```
```suggestion
For this exercise, assume that the VAV has a nominal flow rate of <i>100 m<sup>3</sup>/h</i>, which equals <i>0.033 kg/s</i>. A nominal pressure drop of <i>50 Pa</i>
```

annadellisola commented on IDEAS/Resources/ReferenceResults/Dymola/IDEAS_Examples_Tutorial_DetailedHouse_DetailedHouse1.txt:
Increase of "number of continuous time states" due to the use of two different components for ceiling and SlabOnGround, and the change from **IDEAS.Buildings.Validation.Data.Constructions.HeavyFloor** to **IDEAS.Buildings.Data.Constructions.FloorOnGround**.
<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:dt="uuid:C2F41010-65B3-11d1-A29F-00AA00C14882"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=OneNote.File>
<meta name=Generator content="Microsoft OneNote 15">
</head>

<body lang=en-GB style='font-family:Calibri;font-size:11.0pt'>
<!--StartFragment-->

<div style='direction:ltr'>


Update   Ref Results | Old   Reference Results
-- | --
cei.layMul.monLay[1].monLayDyn.T[1]   cei.layMul.monLay[1].monLayDyn.T[2]   cei.layMul.monLay[2].monLayDyn.T[1]   cei.layMul.monLay[3].monLayDyn.T[1]   outWalEas.layMul.monLay[1].monLayDyn.T[1]   outWalEas.layMul.monLay[1].monLayDyn.T[2]   outWalEas.layMul.monLay[2].monLayDyn.T[1]   outWalEas.layMul.monLay[3].monLayDyn.T[1]   outWalEas.layMul.monLay[3].monLayDyn.T[2]   outWalNor.layMul.monLay[1].monLayDyn.T[1]   outWalNor.layMul.monLay[1].monLayDyn.T[2]   outWalNor.layMul.monLay[2].monLayDyn.T[1]   outWalNor.layMul.monLay[3].monLayDyn.T[1]   outWalNor.layMul.monLay[3].monLayDyn.T[2]   outWalSou.layMul.monLay[1].monLayDyn.T[1]   outWalSou.layMul.monLay[1].monLayDyn.T[2]   outWalSou.layMul.monLay[2].monLayDyn.T[1]   outWalSou.layMul.monLay[3].monLayDyn.T[1]   outWalSou.layMul.monLay[3].monLayDyn.T[2]   outWalWes.layMul.monLay[1].monLayDyn.T[1]   outWalWes.layMul.monLay[1].monLayDyn.T[2]   outWalWes.layMul.monLay[2].monLayDyn.T[1]   outWalWes.layMul.monLay[3].monLayDyn.T[1]   outWalWes.layMul.monLay[3].monLayDyn.T[2]   slaOnGro.layGro.monLay[1].monLayDyn.T[1]   slaOnGro.layGro.monLay[1].monLayDyn.T[2]   slaOnGro.layGro.monLay[1].monLayDyn.T[3]   slaOnGro.layGro.monLay[1].monLayDyn.T[4]   slaOnGro.layGro.monLay[1].monLayDyn.T[5]   slaOnGro.layGro.monLay[1].monLayDyn.T[6]   slaOnGro.layGro.monLay[1].monLayDyn.T[7]   slaOnGro.layGro.monLay[2].monLayDyn.T[1]   slaOnGro.layGro.monLay[2].monLayDyn.T[2]   slaOnGro.layGro.monLay[2].monLayDyn.T[3]   slaOnGro.layGro.monLay[2].monLayDyn.T[4]   slaOnGro.layGro.monLay[3].monLayDyn.T[2]   slaOnGro.layMul.monLay[1].monLayDyn.T[1]   slaOnGro.layMul.monLay[1].monLayDyn.T[2]   slaOnGro.layMul.monLay[1].monLayDyn.T[3]   slaOnGro.layMul.monLay[1].monLayDyn.T[4]   slaOnGro.layMul.monLay[2].monLayDyn.T[1]   slaOnGro.layMul.monLay[3].monLayDyn.T[1]   slaOnGro.layMul.monLay[4].monLayDyn.T[1]   win.heaCapGlaExt.T   win.heaCapGlaInt.T   zon.airModel.vol.dynBal.m   zon.airModel.vol.dynBal.medium.Xi[1]   zon.airModel.vol.dynBal.U | floor.layMul.monLay[1].monLayDyn.T[1]   floor.layMul.monLay[1].monLayDyn.T[2]   floor.layMul.monLay[2].monLayDyn.T[1]   outerWall.layMul.monLay[1].monLayDyn.T[1]   outerWall.layMul.monLay[1].monLayDyn.T[2]   outerWall.layMul.monLay[2].monLayDyn.T[1]   outerWall.layMul.monLay[3].monLayDyn.T[1]   outerWall.layMul.monLay[3].monLayDyn.T[2]   outerWall1.layMul.monLay[1].monLayDyn.T[1]   outerWall1.layMul.monLay[1].monLayDyn.T[2]   outerWall1.layMul.monLay[2].monLayDyn.T[1]   outerWall1.layMul.monLay[3].monLayDyn.T[1]   outerWall1.layMul.monLay[3].monLayDyn.T[2]   outerWall2.layMul.monLay[1].monLayDyn.T[1]   outerWall2.layMul.monLay[1].monLayDyn.T[2]   outerWall2.layMul.monLay[2].monLayDyn.T[1]   outerWall2.layMul.monLay[3].monLayDyn.T[1]   outerWall2.layMul.monLay[3].monLayDyn.T[2]   outerWall3.layMul.monLay[1].monLayDyn.T[1]   outerWall3.layMul.monLay[1].monLayDyn.T[2]   outerWall3.layMul.monLay[2].monLayDyn.T[1]   outerWall3.layMul.monLay[3].monLayDyn.T[1]   outerWall3.layMul.monLay[3].monLayDyn.T[2]   window.heaCapGlaExt.T   window.heaCapGlaInt.T   zone.airModel.vol.dynBal.m   zone.airModel.vol.dynBal.medium.Xi[1]   zone.airModel.vol.dynBal.U



</div>

<!--EndFragment-->
</body>

</html>


The figure below shows the difference in the reference variables:
![image](https://github.com/user-attachments/assets/5e6f9ed5-9db6-45d7-b6ca-7e25cdeb312c)
Ref_res:  `InternalWall` for ceiling and floor with highly insulated BESTEST heavy wall as construction type 

Test: distinct `SlabOnGround` with FloorOnGround construction type, and ceiling lead to higher heat losses

annadellisola commented on IDEAS/Resources/ReferenceResults/Dymola/IDEAS_Examples_Tutorial_DetailedHouse_DetailedHouse3.txt:
Same considerations of DetailedHouse1 holds for the change in the reference results. 
The figure below shows the change in reference variables:
![image](https://github.com/user-attachments/assets/5bb27e6f-f533-49dc-a2c8-3a84863160d2)

annadellisola commented on IDEAS/Resources/ReferenceResults/Dymola/IDEAS_Examples_Tutorial_DetailedHouse_DetailedHouse6.txt:
Increase of "number of continuous time states" due to the use of two different components for ceiling and SlabOnGround, the change from IDEAS.Buildings.Validation.Data.Constructions.HeavyFloor to IDEAS.Buildings.Data.Constructions.FloorOnGround in both zones and the addition of `pumEmi`:
<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:dt="uuid:C2F41010-65B3-11d1-A29F-00AA00C14882"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=OneNote.File>
<meta name=Generator content="Microsoft OneNote 15">
</head>

<body lang=en-GB style='font-family:Calibri;font-size:11.0pt'>
<!--StartFragment-->

<div style='direction:ltr'>


Update   Ref Results | Old   Reference Results
-- | --
EEl.y   heaPum.con.vol.dynBal.U   heaPum.eva.vol.dynBal.U   pumEmi.motSpe.y   pumEmi.vol.dynBal.U   pumPri.motSpe.y   pumPri.vol.dynBal.U   pumSec.motSpe.y   pumSec.vol.dynBal.U   rad.vol[1].dynBal.U   rad.vol[2].dynBal.U   rad.vol[3].dynBal.U   rad.vol[4].dynBal.U   rad.vol[5].dynBal.U   rad1.vol[1].dynBal.U   rad1.vol[2].dynBal.U   rad1.vol[3].dynBal.U   rad1.vol[4].dynBal.U   rad1.vol[5].dynBal.U   rectangularZoneTemplate.airModel.vol.dynBal.m   rectangularZoneTemplate.airModel.vol.dynBal.medium.Xi[1]   rectangularZoneTemplate.airModel.vol.dynBal.U   rectangularZoneTemplate.intC.layMul.monLay[1].monLayDyn.T[1]   rectangularZoneTemplate.intC.layMul.monLay[1].monLayDyn.T[2]   rectangularZoneTemplate.intC.layMul.monLay[2].monLayDyn.T[1]   rectangularZoneTemplate.intC.layMul.monLay[3].monLayDyn.T[1]   rectangularZoneTemplate.intC.layMul.monLay[3].monLayDyn.T[2]   rectangularZoneTemplate.outA.layMul.monLay[1].monLayDyn.T[1]   rectangularZoneTemplate.outA.layMul.monLay[1].monLayDyn.T[2]   rectangularZoneTemplate.outA.layMul.monLay[2].monLayDyn.T[1]   rectangularZoneTemplate.outA.layMul.monLay[3].monLayDyn.T[1]   rectangularZoneTemplate.outA.layMul.monLay[3].monLayDyn.T[2]   rectangularZoneTemplate.outB.layMul.monLay[1].monLayDyn.T[1]   rectangularZoneTemplate.outB.layMul.monLay[1].monLayDyn.T[2]   rectangularZoneTemplate.outB.layMul.monLay[2].monLayDyn.T[1]   rectangularZoneTemplate.outB.layMul.monLay[3].monLayDyn.T[1]   rectangularZoneTemplate.outB.layMul.monLay[3].monLayDyn.T[2]   rectangularZoneTemplate.outCei.layMul.monLay[1].monLayDyn.T[1]   rectangularZoneTemplate.outCei.layMul.monLay[1].monLayDyn.T[2]   rectangularZoneTemplate.outCei.layMul.monLay[2].monLayDyn.T[1]   rectangularZoneTemplate.outCei.layMul.monLay[3].monLayDyn.T[1]   rectangularZoneTemplate.outD.layMul.monLay[1].monLayDyn.T[1]   rectangularZoneTemplate.outD.layMul.monLay[1].monLayDyn.T[2]   rectangularZoneTemplate.outD.layMul.monLay[2].monLayDyn.T[1]   rectangularZoneTemplate.outD.layMul.monLay[3].monLayDyn.T[1]   rectangularZoneTemplate.outD.layMul.monLay[3].monLayDyn.T[2]   rectangularZoneTemplate.slaOnGro.layGro.monLay[1].monLayDyn.T[1]   rectangularZoneTemplate.slaOnGro.layGro.monLay[1].monLayDyn.T[2]   rectangularZoneTemplate.slaOnGro.layGro.monLay[1].monLayDyn.T[3]   rectangularZoneTemplate.slaOnGro.layGro.monLay[1].monLayDyn.T[4]   rectangularZoneTemplate.slaOnGro.layGro.monLay[1].monLayDyn.T[5]   rectangularZoneTemplate.slaOnGro.layGro.monLay[1].monLayDyn.T[6]   rectangularZoneTemplate.slaOnGro.layGro.monLay[1].monLayDyn.T[7]   rectangularZoneTemplate.slaOnGro.layGro.monLay[2].monLayDyn.T[1]   rectangularZoneTemplate.slaOnGro.layGro.monLay[2].monLayDyn.T[2]   rectangularZoneTemplate.slaOnGro.layGro.monLay[2].monLayDyn.T[3]   rectangularZoneTemplate.slaOnGro.layGro.monLay[2].monLayDyn.T[4]   rectangularZoneTemplate.slaOnGro.layGro.monLay[3].monLayDyn.T[2]   rectangularZoneTemplate.slaOnGro.layMul.monLay[1].monLayDyn.T[1]   rectangularZoneTemplate.slaOnGro.layMul.monLay[1].monLayDyn.T[2]   rectangularZoneTemplate.slaOnGro.layMul.monLay[1].monLayDyn.T[3]   rectangularZoneTemplate.slaOnGro.layMul.monLay[1].monLayDyn.T[4]   rectangularZoneTemplate.slaOnGro.layMul.monLay[2].monLayDyn.T[1]   rectangularZoneTemplate.slaOnGro.layMul.monLay[3].monLayDyn.T[1]   rectangularZoneTemplate.slaOnGro.layMul.monLay[4].monLayDyn.T[1]   rectangularZoneTemplate.winA.heaCapGlaExt.T   rectangularZoneTemplate.winA.heaCapGlaInt.T   rectangularZoneTemplate1.airModel.vol.dynBal.m   rectangularZoneTemplate1.airModel.vol.dynBal.medium.Xi[1]   rectangularZoneTemplate1.airModel.vol.dynBal.U   rectangularZoneTemplate1.outB.layMul.monLay[1].monLayDyn.T[1]   rectangularZoneTemplate1.outB.layMul.monLay[1].monLayDyn.T[2]   rectangularZoneTemplate1.outB.layMul.monLay[2].monLayDyn.T[1]   rectangularZoneTemplate1.outB.layMul.monLay[3].monLayDyn.T[1]   rectangularZoneTemplate1.outB.layMul.monLay[3].monLayDyn.T[2]   rectangularZoneTemplate1.outC.layMul.monLay[1].monLayDyn.T[1]   rectangularZoneTemplate1.outC.layMul.monLay[1].monLayDyn.T[2]   rectangularZoneTemplate1.outC.layMul.monLay[2].monLayDyn.T[1]   rectangularZoneTemplate1.outC.layMul.monLay[3].monLayDyn.T[1]   rectangularZoneTemplate1.outC.layMul.monLay[3].monLayDyn.T[2]   rectangularZoneTemplate1.outCei.layMul.monLay[1].monLayDyn.T[1]   rectangularZoneTemplate1.outCei.layMul.monLay[1].monLayDyn.T[2]   rectangularZoneTemplate1.outCei.layMul.monLay[2].monLayDyn.T[1]   rectangularZoneTemplate1.outCei.layMul.monLay[3].monLayDyn.T[1]   rectangularZoneTemplate1.outD.layMul.monLay[1].monLayDyn.T[1]   rectangularZoneTemplate1.outD.layMul.monLay[1].monLayDyn.T[2]   rectangularZoneTemplate1.outD.layMul.monLay[2].monLayDyn.T[1]   rectangularZoneTemplate1.outD.layMul.monLay[3].monLayDyn.T[1]   rectangularZoneTemplate1.outD.layMul.monLay[3].monLayDyn.T[2]   rectangularZoneTemplate1.slaOnGro.layGro.monLay[1].monLayDyn.T[1]   rectangularZoneTemplate1.slaOnGro.layGro.monLay[1].monLayDyn.T[2]   rectangularZoneTemplate1.slaOnGro.layGro.monLay[1].monLayDyn.T[3]   rectangularZoneTemplate1.slaOnGro.layGro.monLay[1].monLayDyn.T[4]   rectangularZoneTemplate1.slaOnGro.layGro.monLay[1].monLayDyn.T[5]   rectangularZoneTemplate1.slaOnGro.layGro.monLay[1].monLayDyn.T[6]   rectangularZoneTemplate1.slaOnGro.layGro.monLay[1].monLayDyn.T[7]   rectangularZoneTemplate1.slaOnGro.layGro.monLay[2].monLayDyn.T[1]   rectangularZoneTemplate1.slaOnGro.layGro.monLay[2].monLayDyn.T[2]   rectangularZoneTemplate1.slaOnGro.layGro.monLay[2].monLayDyn.T[3]   rectangularZoneTemplate1.slaOnGro.layGro.monLay[2].monLayDyn.T[4]   rectangularZoneTemplate1.slaOnGro.layGro.monLay[3].monLayDyn.T[2]   rectangularZoneTemplate1.slaOnGro.layMul.monLay[1].monLayDyn.T[1]   rectangularZoneTemplate1.slaOnGro.layMul.monLay[1].monLayDyn.T[2]   rectangularZoneTemplate1.slaOnGro.layMul.monLay[1].monLayDyn.T[3]   rectangularZoneTemplate1.slaOnGro.layMul.monLay[1].monLayDyn.T[4]   rectangularZoneTemplate1.slaOnGro.layMul.monLay[2].monLayDyn.T[1]   rectangularZoneTemplate1.slaOnGro.layMul.monLay[3].monLayDyn.T[1]   rectangularZoneTemplate1.slaOnGro.layMul.monLay[4].monLayDyn.T[1]   rectangularZoneTemplate1.winC.heaCapGlaExt.T   rectangularZoneTemplate1.winC.heaCapGlaInt.T   senTemSup.T   tan.vol[1].dynBal.U   tan.vol[2].dynBal.U   val.val.actPos.y   val1.val.actPos.y | EEl.y   heaPum.con.vol.dynBal.U   heaPum.eva.vol.dynBal.U   pumpPrim.motSpe.y   pumpPrim.vol.dynBal.U   pumpSec.motSpe.y   pumpSec.vol.dynBal.U   rad.vol[1].dynBal.U   rad.vol[2].dynBal.U   rad.vol[3].dynBal.U   rad.vol[4].dynBal.U   rad.vol[5].dynBal.U   rad1.vol[1].dynBal.U   rad1.vol[2].dynBal.U   rad1.vol[3].dynBal.U   rad1.vol[4].dynBal.U   rad1.vol[5].dynBal.U   rectangularZoneTemplate.airModel.vol.dynBal.m   rectangularZoneTemplate.airModel.vol.dynBal.medium.Xi[1]   rectangularZoneTemplate.airModel.vol.dynBal.U   rectangularZoneTemplate.intC.layMul.monLay[1].monLayDyn.T[1]   rectangularZoneTemplate.intC.layMul.monLay[1].monLayDyn.T[2]   rectangularZoneTemplate.intC.layMul.monLay[2].monLayDyn.T[1]   rectangularZoneTemplate.intC.layMul.monLay[3].monLayDyn.T[1]   rectangularZoneTemplate.intC.layMul.monLay[3].monLayDyn.T[2]   rectangularZoneTemplate.intFlo.layMul.monLay[1].monLayDyn.T[1]   rectangularZoneTemplate.intFlo.layMul.monLay[1].monLayDyn.T[2]   rectangularZoneTemplate.intFlo.layMul.monLay[2].monLayDyn.T[1]   rectangularZoneTemplate.intFlo.layMul.monLay[3].monLayDyn.T[1]   rectangularZoneTemplate.intFlo.layMul.monLay[3].monLayDyn.T[2]   rectangularZoneTemplate.outA.layMul.monLay[1].monLayDyn.T[1]   rectangularZoneTemplate.outA.layMul.monLay[1].monLayDyn.T[2]   rectangularZoneTemplate.outA.layMul.monLay[2].monLayDyn.T[1]   rectangularZoneTemplate.outA.layMul.monLay[3].monLayDyn.T[1]   rectangularZoneTemplate.outA.layMul.monLay[3].monLayDyn.T[2]   rectangularZoneTemplate.outB.layMul.monLay[1].monLayDyn.T[1]   rectangularZoneTemplate.outB.layMul.monLay[1].monLayDyn.T[2]   rectangularZoneTemplate.outB.layMul.monLay[2].monLayDyn.T[1]   rectangularZoneTemplate.outB.layMul.monLay[3].monLayDyn.T[1]   rectangularZoneTemplate.outB.layMul.monLay[3].monLayDyn.T[2]   rectangularZoneTemplate.outD.layMul.monLay[1].monLayDyn.T[1]   rectangularZoneTemplate.outD.layMul.monLay[1].monLayDyn.T[2]   rectangularZoneTemplate.outD.layMul.monLay[2].monLayDyn.T[1]   rectangularZoneTemplate.outD.layMul.monLay[3].monLayDyn.T[1]   rectangularZoneTemplate.outD.layMul.monLay[3].monLayDyn.T[2]   rectangularZoneTemplate.winA.heaCapGlaExt.T   rectangularZoneTemplate.winA.heaCapGlaInt.T   rectangularZoneTemplate1.airModel.vol.dynBal.m   rectangularZoneTemplate1.airModel.vol.dynBal.medium.Xi[1]   rectangularZoneTemplate1.airModel.vol.dynBal.U   rectangularZoneTemplate1.intFlo.layMul.monLay[1].monLayDyn.T[1]   rectangularZoneTemplate1.intFlo.layMul.monLay[1].monLayDyn.T[2]   rectangularZoneTemplate1.intFlo.layMul.monLay[2].monLayDyn.T[1]   rectangularZoneTemplate1.intFlo.layMul.monLay[3].monLayDyn.T[1]   rectangularZoneTemplate1.intFlo.layMul.monLay[3].monLayDyn.T[2]   rectangularZoneTemplate1.outB.layMul.monLay[1].monLayDyn.T[1]   rectangularZoneTemplate1.outB.layMul.monLay[1].monLayDyn.T[2]   rectangularZoneTemplate1.outB.layMul.monLay[2].monLayDyn.T[1]   rectangularZoneTemplate1.outB.layMul.monLay[3].monLayDyn.T[1]   rectangularZoneTemplate1.outB.layMul.monLay[3].monLayDyn.T[2]   rectangularZoneTemplate1.outC.layMul.monLay[1].monLayDyn.T[1]   rectangularZoneTemplate1.outC.layMul.monLay[1].monLayDyn.T[2]   rectangularZoneTemplate1.outC.layMul.monLay[2].monLayDyn.T[1]   rectangularZoneTemplate1.outC.layMul.monLay[3].monLayDyn.T[1]   rectangularZoneTemplate1.outC.layMul.monLay[3].monLayDyn.T[2]   rectangularZoneTemplate1.outD.layMul.monLay[1].monLayDyn.T[1]   rectangularZoneTemplate1.outD.layMul.monLay[1].monLayDyn.T[2]   rectangularZoneTemplate1.outD.layMul.monLay[2].monLayDyn.T[1]   rectangularZoneTemplate1.outD.layMul.monLay[3].monLayDyn.T[1]   rectangularZoneTemplate1.outD.layMul.monLay[3].monLayDyn.T[2]   rectangularZoneTemplate1.winC.heaCapGlaExt.T   rectangularZoneTemplate1.winC.heaCapGlaInt.T   senTemSup.T   tan.vol[1].dynBal.U   tan.vol[2].dynBal.U   val.val.actPos.y   val1.val.actPos.y



</div>

<!--EndFragment-->
</body>

</html>

The figure below shows the change in reference variables:
![image](https://github.com/user-attachments/assets/069dc184-ae06-4799-bd73-f5124f78ce74)
In the Test results the TSensor is lower due to change in floor construction type.
Less on/off cycle of the heat pump occur due to the decoupling of production and emission system through the tank.

annadellisola commented on IDEAS/Resources/ReferenceResults/Dymola/IDEAS_Examples_Tutorial_DetailedHouse_DetailedHouse7.txt:
Same considerations of DetailedHouse1 holds for the change in the reference results.
The figure below shows the change in reference variables:
![image](https://github.com/user-attachments/assets/05be6e5d-56d3-4a06-a849-3f42ca44c246)
In the Test results the TSensor is lower due to change in floor construction type.
Less on/off cycle of the heat pump and a more stable condenser temperature results from decoupling the production and emission system through the tank.
